src/app/store/students/students.reducer.ts

```diff
@@ -51,5 +53,32 @@ export const studentsReducer = createReducer(
       hasStudentAddFailed: true,
       error: action.error
     };
+  }),
+  on(loadStudents, (state) => {
+    return {
+      ...state,
+      loading: true,
+      loaded: false,
+      hasStudentLoadFailed: false,
```

Comment:
hasStudentLoadFailed has already a default value of false. It can be skipped in places, where it is repeated, like on loadStudents and loadStudentsSuccess :)